common/Resources/ti.internal/extensions/node/internal/util/inspect.js

```diff
@@ -0,0 +1,1686 @@
+	// Otherwise we'd print this twice.
+	if (typeof tag !== 'string'
+			|| tag !== ''
+			&& (ctx.showHidden ? hasOwnProperty : propertyIsEnumerable)(
+				value, Symbol.toStringTag
+			)) {
+		tag = '';
+	}
+	let base = '';
+	let formatter = getEmptyFormatArray;
+	let braces;
+	let noIterator = true;
+	let i = 0;
+	const filter = ctx.showHidden ? ALL_PROPERTIES : ONLY_ENUMERABLE;
+
+	let extrasType = kObjectType;
+
+	// Iterators and the rest are split to reduce checks.
+	if (value[Symbol.iterator]) {
+		noIterator = false;
+		if (Array.isArray(value)) {
+			keys = getOwnNonIndexProperties(value, filter);
+			// Only set the constructor for non ordinary ("Array [...]") arrays.
+			const prefix = getPrefix(constructor, tag, 'Array');
+			braces = [ `${prefix === 'Array ' ? '' : prefix}[`, ']' ];
+			if (value.length === 0 && keys.length === 0) {
+				return `${braces[0]}]`;
+			}
+			extrasType = kArrayExtrasType;
+			formatter = formatArray;
+		} else if (isSet(value)) {
+			keys = getKeys(value, ctx.showHidden);
+			const prefix = getPrefix(constructor, tag, 'Set');
+			if (value.size === 0 && keys.length === 0) {
+				return `${prefix}{}`;
+			}
+			braces = [ `${prefix}{`, '}' ];
+			formatter = formatSet;
+		} else if (isMap(value)) {
+			keys = getKeys(value, ctx.showHidden);
+			const prefix = getPrefix(constructor, tag, 'Map');
+			if (value.size === 0 && keys.length === 0) {
+				return `${prefix}{}`;
+			}
+			braces = [ `${prefix}{`, '}' ];
+			formatter = formatMap;
+		} else if (isTypedArray(value)) {
+			keys = getOwnNonIndexProperties(value, filter);
+			const prefix = constructor !== null
+				? getPrefix(constructor, tag)
+				: getPrefix(constructor, tag, findTypedConstructor(value).name);
+			braces = [ `${prefix}[`, ']' ];
+			if (value.length === 0 && keys.length === 0 && !ctx.showHidden) {
+				return `${braces[0]}]`;
+			}
+			formatter = formatTypedArray;
+			extrasType = kArrayExtrasType;
+		} else if (isMapIterator(value)) {
+			keys = getKeys(value, ctx.showHidden);
+			braces = getIteratorBraces('Map', tag);
+			formatter = formatIterator;
+		} else if (isSetIterator(value)) {
+			keys = getKeys(value, ctx.showHidden);
+			braces = getIteratorBraces('Set', tag);
+			formatter = formatIterator;
+		} else {
+			noIterator = true;
+		}
+	}
+
+	if (noIterator) {
+		keys = getKeys(value, ctx.showHidden);
+		braces = [ '{', '}' ];
+		if (constructor === 'Object') {
+			if (isArgumentsObject(value)) {
+				braces[0] = '[Arguments] {';
+			} else if (tag !== '') {
+				braces[0] = `${getPrefix(constructor, tag, 'Object')}{`;
+			}
+			if (keys.length === 0) {
+				return `${braces[0]}}`;
+			}
+		} else if (typeof value === 'function') {
+			base = getFunctionBase(value, constructor, tag);
+			if (keys.length === 0) {
+				return ctx.stylize(base, 'special');
+			}
+		} else if (isRegExp(value)) {
+			// Make RegExps say that they are RegExps
+			const regExp = constructor !== null ? value : new RegExp(value);
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/internal/util/inspect.js;
  Line: 671;
-->

- :warning: common/Resources/ti.internal/extensions/node/internal/util/inspect.js line 671 – Found non-literal argument to RegExp Constructor (security/detect-non-literal-regexp)